### PR TITLE
chapi2: Add chapiclient package

### PR DESCRIPTION
**Pull request by NimbleMike**
* Problem:
  * Need to port CHAPI1 client package to CHAPI2
* Implementation:
  * handler_windows.go
    * Changed chapiPortFileName to ChapiPortFileName so that string is accessible to other packages (e.g. CHAPI client)
  * chapiclient.go
    * Ported CHAPI1 code to CHAPI2
    * NewChapiClient() and NewChapiClientWithTimeout() are platform independent methods to create a new CHAPI Client object
    * **Only** wrap CHAPI2 endpoints in this module
    * Simplified all endpoint handlers by wrapping chapiClient.client.DoJSON in a new chapiClient.chapiClientDoJSON() function (shared/common error handling)
    * Updated Response object to expect a cerrors.ChapiError object instead of a generic error
    * Compilation will fail if Client object does not support CHAPI2's Driver interface (i.e. keep parity between client and server)
  * chapiclient_linux.go
    * Provide Linux specific Client object creation methods
    * All Socket methods are in this Linux module since they're not applicable to Windows
  * chapiclient_windows.go
    * Provide Windows specific Client object creation methods
    * All HTTP port methods are in this Windows module since they're not applicable to Linux
  * chapiclient_util.go
    * Created a util stub module to add any support routines that are intended to extend the capabilities beyond the CHAPI endpoints
* Testing:
  * Successfully built Linux and Windows binaries
  * Created a chapid resident service for Windows and exercised available endpoints using chapiclient package
  * Linux testing and final implementation pending
* Reviewers:
  * @shivamerla, @suneeth51
* Bug: N/A

### CODE REVIEW

File: chapi2/chapiclient/chapiclient.go [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+	// CHAPI request was successful; return the HTTP status code with no error
+	return statusCode, nil
+}
+
+// appendQuerySerialNumber appends a serial number query to the given URI
+func (chapiClient *Client) appendQuerySerialNumber(uri string, serialNumber string) string {
+	return chapiClient.appendQuery(uri, querySerialNumber, serialNumber)
+}
+
+// appendQuerySerialNumber appends a mount ID query to the given URI
```
Comment (shivamerla) [inline chapi2/chapiclient/chapiclient.go]: nit: typo in function name..

File: chapi2/chapiclient/chapiclient.go [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+func (chapiClient *Client) appendQueryMountPointID(uri string, serialNumber string) string {
```
Comment (shivamerla) [inline chapi2/chapiclient/chapiclient.go]: mountID needs to be passed here?

File: chapi2/chapiclient/chapiclient.go (hunk 1 above)
Comment (NimbleMike) [reply chapi2/chapiclient/chapiclient.go]: Thanks for catching that.  Fixing it.

**Comment by NimbleMike** [comment]
Fixed function name in comment plus function parameter name.  No functional changes, just corrected declaration and comment.

**Pull request merged**